osu.Game.Rulesets.Osu/Difficulty/Evaluators/FlashlightEvaluator.cs

```diff
@@ -73,6 +79,30 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool hidd
             if (hidden)
                 result *= 1.0 + hidden_bonus;
 
+            double sliderBonus = 0.0;
+
+            if (osuCurrent.BaseObject is Slider)
+            {
+                Debug.Assert(osuCurrent.TravelTime > 0);
+
+                Slider osuSlider = (Slider)(osuCurrent.BaseObject);
+
+                // Invert the scaling factor to determine the true travel distance independent of circle size.
+                double pixelTravelDistance = osuCurrent.TravelDistance / scalingFactor;
+
+                // Reward sliders based on velocity.
+                sliderBonus = Math.Pow(Math.Max(0.0, pixelTravelDistance / osuCurrent.TravelTime - min_velocity), 0.5);
+
+                // Longer sliders require more memorisation.
+                sliderBonus *= pixelTravelDistance;
+
+                // Reward sliders with repeats.
+                if (osuSlider.RepeatCount > 0)
+                    sliderBonus *= repeat_bonus;
```

Comment:
A 3x repeat bonus, in Aim, is achieved with ~30 repeats. This immediately signals to me that this bonus is weighted too highly currently: either by `repeat_bonus` being too high, or by the full bonus being applied by only one repeat. Is this balanced around a specific map/case?

Reply:
So I was misunderstanding how `TravelDistance` worked. StanR originally pointed out the issue, linking me a map with a long slider with many repeats having a large spike in FL strain. I originally thought the issue was because of the newly added repeat bonus to `TravelDistance`, so I removed that and gave a flat bonus to all repeat sliders.

That wasn't the issue, rather the issue was that `TravelDistance` includes the distance traveled by repeats (which I didn't think was included).

I solved this by taking the slider bonus and dividing it by `Slider.RepeatCount + 1`, which should still give repeat sliders a bonus compared to normal sliders, but removes the possibility of abuse.